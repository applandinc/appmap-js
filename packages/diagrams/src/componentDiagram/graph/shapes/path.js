import { line, curveBasis } from 'd3-shape';
import { createSVGElement } from '../util';

const lineGenerator = line().curve(curveBasis);

export default class Path {
  constructor(points) {
    this.element = createSVGElement('path');
    this.element.setAttribute('d', lineGenerator(points));
  }

  setPoints(points) {
    this.element.setAttribute('d', lineGenerator(points));
  }
}
