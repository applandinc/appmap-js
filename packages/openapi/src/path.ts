import { OpenAPIV3 } from 'openapi-types';
import Method from './method';
import { RPCRequest } from './rpcRequest';
import Warnings, { WarningProvider } from './Warnings';

export default class Path {
  methods: Record<string, Method> = {};
  warnings = new Warnings();

  get containedWarningProviders(): WarningProvider[] {
    return Object.values(this.methods);
  }

  openapi(): OpenAPIV3.PathItemObject {
    return Object.keys(this.methods)
      .sort()
      .reduce((memo, methodName: string) => {
        const method = this.methods[methodName as OpenAPIV3.HttpMethods];
        const methodOpenAPI = method.openapi();
        // Don't emit empty methods.
        if (Object.keys(methodOpenAPI.responses).length) {
          // eslint-disable-next-line no-param-reassign
          memo[methodName as OpenAPIV3.HttpMethods] = methodOpenAPI;
        }
        return memo;
      }, {} as Record<OpenAPIV3.HttpMethods, OpenAPIV3.OperationObject>) as OpenAPIV3.PathItemObject;
  }

  addRpcRequest(rpcRequest: RPCRequest): void {
    const method = rpcRequest.requestMethod;
    if (!method) {
      return;
    }

    if (!this.methods[method]) {
      this.methods[method] = new Method();
    }
    const methodObj = this.methods[method];
    methodObj.addRpcRequest(rpcRequest);
  }
}
