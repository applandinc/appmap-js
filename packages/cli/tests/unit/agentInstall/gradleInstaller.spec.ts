import path from 'path';
import glob from 'glob';
import sinon from 'sinon';
import inquirer from 'inquirer';
import GradleInstaller from '../../../src/cmds/agentInstaller/gradleInstaller';
import EncodedFile from '../../../src/encodedFile';
import InstallerUI from '../../../src/cmds/agentInstaller/installerUI';

const fixtureDir = path.join(__dirname, '..', 'fixtures', 'java', 'gradle');
const dataDir = path.join(fixtureDir, 'data');

const interactiveUI = new InstallerUI(true, {});

describe('GradleInstaller', () => {
  afterEach(() => {
    sinon.restore();
  });

  describe('available', () => {
    it('detects a gradle project', () => {
      const projectDirectory = path.join(fixtureDir, 'example-project');
      const maven = new GradleInstaller(projectDirectory);
      expect(maven.available()).resolves.toBe(true);
    });
  });

  describe('installAgent', () => {
    ['.gradle', '.gradle.kts'].forEach((testExt) => {
      const files = glob.sync(path.join(dataDir, `*${testExt}`));

      files.forEach((file) => {
        const test = path.basename(file, testExt);

        // By default, we'll run all test. This is here to make it easy to skip
        // them by name: change true to false, and check for the test name of
        // interest.
        const testFn = true || test === 'extra-repo-block' ? it : xit;

        const actualFile = `${test}${testExt}`;

        testFn(`transforms ${actualFile} as expected`, async () => {
          const gradle = new GradleInstaller('.');

          sinon
            .stub(inquirer, 'prompt')
            .resolves({ addMavenCentral: 'Yes', userWillContinue: 'Continue' });

          sinon.stub(gradle, 'identifyGradleFile').value(actualFile);

          sinon.stub(gradle, 'buildFilePath').value(path.join(dataDir, actualFile));

          const efWrite = sinon.stub(EncodedFile.prototype, 'write');

          await gradle.installAgent(interactiveUI);

          let actual = efWrite.getCall(-1).args[0].toString();
          expect(actual).toMatchSnapshot();
        });
      });
    });
  });
});
