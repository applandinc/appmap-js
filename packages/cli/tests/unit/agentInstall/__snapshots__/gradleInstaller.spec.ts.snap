// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GradleInstaller installAgent transforms buildscript.gradle as expected 1`] = `
"buildscript {
    repositories {
        mavenCentral()
    }
}

plugins {
    id 'com.appland.appmap' version '1.1.0'
}

repositories {
    mavenCentral()
}


application {
    mainClass = 'demo.App'
}
"
`;

exports[`GradleInstaller installAgent transforms buildscript.gradle.kts as expected 1`] = `
"buildscript {
    repositories {
        mavenCentral()
    }
}

plugins {
    id(\\"com.appland.appmap\\") version \\"1.1.0\\"
}

repositories {
    mavenCentral()
}


application {
    mainClass = 'demo.App'
}
"
`;

exports[`GradleInstaller installAgent transforms buildscript-plugins.gradle as expected 1`] = `
"buildscript {
}

plugins {
  id 'com.appland.appmap' version '1.1.0'
}

repositories {
  mavenCentral()
}
"
`;

exports[`GradleInstaller installAgent transforms buildscript-plugins.gradle.kts as expected 1`] = `
"buildscript {
}

plugins {
  id(\\"com.appland.appmap\\") version \\"1.1.0\\"
}

repositories {
  mavenCentral()
}
"
`;

exports[`GradleInstaller installAgent transforms empty.gradle as expected 1`] = `
"/*
 * This file was generated by the Gradle 'init' task.
 *
 * This is a general purpose Gradle build.
 * Learn more about Gradle by exploring our samples at https://docs.gradle.org/7.2/samples
 */


plugins {
  id 'com.appland.appmap' version '1.1.0'
}

repositories {
  mavenCentral()
}
"
`;

exports[`GradleInstaller installAgent transforms empty.gradle.kts as expected 1`] = `
"/*
 * This file was generated by the Gradle 'init' task.
 *
 * This is a general purpose Gradle build.
 * Learn more about Gradle by exploring our samples at https://docs.gradle.org/7.2/samples
 */


plugins {
  id(\\"com.appland.appmap\\") version \\"1.1.0\\"
}

repositories {
  mavenCentral()
}
"
`;

exports[`GradleInstaller installAgent transforms existing-appmap.gradle as expected 1`] = `
"plugins {
  id 'com.appland.appmap' version '1.1.0'
}

repositories {
  mavenCentral()
}
"
`;

exports[`GradleInstaller installAgent transforms existing-appmap.gradle.kts as expected 1`] = `
"plugins {
  id(\\"com.appland.appmap\\") version \\"1.1.0\\"
}

repositories {
  mavenCentral()
}
"
`;

exports[`GradleInstaller installAgent transforms extra-repo-block.gradle as expected 1`] = `
"// This is a comment before the plugins block.
plugins {
  id 'java'
  id 'com.appland.appmap' version '1.1.0'
}
repositories {
  mavenCentral()
}


// This is a comment after the plugins block

subprojects {
}

allprojects {
  repositories {
    google()
  }
}
"
`;

exports[`GradleInstaller installAgent transforms extra-repo-block.gradle.kts as expected 1`] = `
"// This is a comment before the plugins block.
plugins {
  java
  id(\\"com.appland.appmap\\") version \\"1.1.0\\"
}
repositories {
  mavenCentral()
}


// This is a comment after the plugins block

subprojects {
}

allprojects {
  repositories {
    google()
  }
}
"
`;

exports[`GradleInstaller installAgent transforms no-maven.gradle as expected 1`] = `
"// This is a comment before the plugins block.
plugins {
  id 'java'
  id 'com.appland.appmap' version '1.1.0'
}

repositories {
  google()
  mavenCentral()
}
"
`;

exports[`GradleInstaller installAgent transforms no-maven.gradle.kts as expected 1`] = `
"// This is a comment before the plugins block.
plugins {
  java
  id(\\"com.appland.appmap\\") version \\"1.1.0\\"
}

repositories {
  google()
  mavenCentral()
}
"
`;

exports[`GradleInstaller installAgent transforms no-plugins.gradle as expected 1`] = `
"

plugins {
  id 'com.appland.appmap' version '1.1.0'
}
repositories {
  mavenCentral()
}
"
`;

exports[`GradleInstaller installAgent transforms no-plugins.gradle.kts as expected 1`] = `
"

plugins {
  id(\\"com.appland.appmap\\") version \\"1.1.0\\"
}
repositories {
  mavenCentral()
}
"
`;

exports[`GradleInstaller installAgent transforms plugins.gradle as expected 1`] = `
"// This is a comment before the plugins block.
plugins {
  id 'java'
  id 'com.appland.appmap' version '1.1.0'
}

repositories {
  mavenCentral()
}
"
`;

exports[`GradleInstaller installAgent transforms plugins.gradle.kts as expected 1`] = `
"// This is a comment before the plugins block.
plugins {
  java
  id(\\"com.appland.appmap\\") version \\"1.1.0\\"
}

repositories {
  mavenCentral()
}
"
`;
